lib/commands/abstract-command.js

```diff
@@ -168,6 +178,31 @@ class AbstractCommand {
         }
     }
 
+    _remindsIfNewVersion(doDebug, callback) {
+        httpClient.request({
+            url: `${CONSTANTS.NPM_REGISTRY_URL_BASE}/${CONSTANTS.APPLICATION_NAME}/latest`,
+            method: CONSTANTS.HTTP_REQUEST.VERB.GET
+        }, 'GET_NPM_REGISTRY', doDebug, (err, response) => {
+            if (err) {
+                Messenger.getInstance().error(`Failed to get the latest version for ${CONSTANTS.APPLICATION_NAME} from NPM registry.\n${err}\n`);
+            } else {
+                const latestVersion = JSON.parse(response.body).version;
+                if (packageJson.version !== latestVersion) {
+                    const semVers = packageJson.version.split('.').map(x => parseInt(x, 10));
+                    const latestSemVers = latestVersion.split('.').map(x => parseInt(x, 10));
+                    if (semVers[0] < latestSemVers[0]) {
+                        Messenger.getInstance().info(`[Info]: \
+New major version (v${latestVersion}) of ${CONSTANTS.APPLICATION_NAME} is available now. Current version v${packageJson.version}. Please update!\n`);
```

Comment:
good future thinking. unfortunately v1 users will not see this message about new major version update. And v2 users will see only when v3 is released. if we do deprecate version using npm, v1 users may see npm warning if they try to re-download v1 https://docs.npmjs.com/cli/deprecate

Reply:
Yep, good to know once it's marked as deprecated, users will see. We will possibly do the same but not sure how long will it happen.